egs/babel_multilang/s5/local/nnet3/run_tdnn_joint_babel_sp_bnf.sh

```diff
@@ -0,0 +1,241 @@
+fi
+feat_dim=`feat-to-dim scp:data/${lang_list[0]}/train${suffix}${aux_suffix}/feats.scp -`
+
+
+if [ $stage -le 9 ]; then
+  mkdir -p $dir/log
+  echo "$0: creating neural net config for multilingual setups"
+   # create the config files for nnet initialization
+  $cmd $dir/log/make_config.log \
+  python steps/nnet3/multi/make_configs.py  \
+    --splice-indexes "$splice_indexes"  \
+    --feat-dim $feat_dim \
+    --ivector-dim $ivector_dim  \
+    --relu-dim $relu_dim \
+    --use-presoftmax-prior-scale false \
+    --num-multiple-targets  "$num_multiple_leaves"  \
+    --add-lda false \
+    --bottleneck-dim $bottleneck_dim --bottleneck-layer $bnf_layer \
+   $dir/configs || exit 1;
+  # Initialize as "raw" nnet, prior to training the LDA-like preconditioning
+  # matrix.  This first config just does any initial splicing that we do;
+  # we do this as it's a convenient way to get the stats for the 'lda-like'
+  # transform.
+  $cmd $dir/log/nnet_init.log \
+    nnet3-init --srand=-2 $dir/configs/init.config $dir/init.raw || exit 1;
+fi
+
+. $dir/configs/vars || exit 1;
+
+if [ $stage -le 10 ]; then
+  for lang in `seq 0 $[$num_lang-1]`; do
+    egs_dir=${multi_egs_dirs[$lang]}
+    ali_dir=${multi_ali_dirs[$lang]}
+    data=data/${lang_list[$lang]}/train${suffix}${aux_suffix}
+    num_frames=$(steps/nnet2/get_num_frames.sh $data)
+    echo num_frames = $num_frames
+    # sets samples_per_iter to have approximately 
+    # same number of archives per language.
+    samples_per_iter=$[$num_frames/($avg_num_archives*$frames_per_eg)]
+    online_ivector_dir=
+    if $use_ivector; then
+      online_ivector_dir=exp/${lang_list[$lang]}/nnet3/ivectors_train${suffix}${ivector_suffix}
+    fi
+    if [ ! -d "$egs_dir" ]; then
+      echo "$0: Generate egs for ${lang_list[$lang]}"
+      if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $egs_dir/storage ]; then
+        utils/create_split_dir.pl \
+         /export/b0{3,4,5,6}/$USER/kaldi-data/egs/${lang_list[$lang]}-$(date +'%m_%d_%H_%M')/s5/$egs_dir/storage $egs_dir/storage
+      fi
+
+      extra_opts=()
+      [ ! -z "$cmvn_opts" ] && extra_opts+=(--cmvn-opts "$cmvn_opts")
+      [ ! -z "$feat_type" ] && extra_opts+=(--feat-type $feat_type)
+      [ ! -z "$online_ivector_dir" ] && extra_opts+=(--online-ivector-dir $online_ivector_dir)
+      extra_opts+=(--left-context $left_context)
+      extra_opts+=(--right-context $right_context)
+      echo "$0: calling get_egs.sh"
+      steps/nnet3/get_egs.sh $egs_opts "${extra_opts[@]}" \
+          --samples-per-iter $samples_per_iter --stage $get_egs_stage \
+          --cmd "$cmd" $egs_opts \
+          --frames-per-eg $frames_per_eg \
+          $data $ali_dir $egs_dir || exit 1;
+      
+    fi
+  done
+fi
+echo print-interval = $print_interval
+if [ $stage -le 11 ]; then
+  echo "$0: training mutilingual model."
+  steps/nnet3/multi/train_tdnn.sh --cmd "$train_cmd" \
```

Comment:
is this script specific to tdnns ? I think most of our training steps are similar for all feed-forward dnns. I recommend renaming it to `steps/nnet3/multilingual/train_dnn.sh`.